maven-di/src/main/java/org/apache/maven/api/di/ProvidesIntoSet.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.api.di;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import org.apache.maven.di.binding.Multibinders;
+import org.apache.maven.di.module.AbstractModule;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * This is a built-in shortcut for {@link Provides provider method} that provides its result as a singleton set
+ * and adds a {@link Multibinders#toSet() set multibinder} for the provided set key to the module.
+ *
+ * @see AbstractModule
+ */
+@Target(METHOD)
+@Retention(RUNTIME)
+public @interface ProvidesIntoSet {}
```

Comment:
guice leak? not sure we need this one

Reply:
I don't think. Planned for removal, that's why it has not been added to `api/maven-api-di` for now.